teuthology/packaging.py

```diff
@@ -643,19 +644,34 @@ def _get_uri_reference(self):
                                                   self.job_config, 'branch')
             sha1 = _get_config_value_for_remote(self.ctx, self.remote,
                                                 self.job_config, 'sha1')
+            ref = None
         else:
-            sha1 = self.sha1
+            ref = self.ref
+            tag = self.tag
             branch = self.branch
+            sha1 = self.sha1
+
+        def warn(attrname):
+            if len([b for b in ref, tag, branch, sha1]) > 1:
+                log.warning(
+                    'More than one of ref, tag, branch, or sha1 supplied; using %s',
+                     attrname
+                )
 
-        if tag:
+        if ref:
```

Comment:
no.  If I were designing from scratch I would have used "ref" and *not* used 'tag' or 'branch', so probably just personal bias ('ref' means 'you know what you're doing')